### ISSUE TITLE
imagemagick mogrify is 32 times slower than graphicmagic

### ISSUE THREAD

**Issue by shirha** [ImageMagick/ImageMagick#2069]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
imagemagick mogrify is 32 times slower than graphicmagic

### Steps to Reproduce
directory with 450 pictures like attached
cvt2.bat
```
FOR /F "tokens=*" %%g IN ('time /T') do (SET TMVAR=%%g)
[magick|gm] mogrify -colors 2 -type bilevel -compress Group4 *.tiff
@echo %TMVAR%
time /T
```
\>`cvt2.cmd` [magick]
12:00 PM
12:32 PM

\>`xcopy ..\bkup . /Y`
\>`cvt2.cmd` [gm]
01:21 PM
01:22 PM

### System Configuration
windows 10
ImageMagick-7.0.10-14-Q16-x64-dll.exe
GraphicsMagick 1.3.35

[page_048.zip](https://github.com/ImageMagick/ImageMagick/files/4684947/page_048.zip)

**Comment by dlemstra** [comment]
I suspect that GraphicsMagick does not dither the image. I get a similar performance with the command:
`magick mogrify +dither -colors 2 -type bilevel -compress Group4 *.tiff`

**Comment by shirha** [comment]
I uninstalled Q16 then installed Q8 and run it again with `+dither`. That improved it a lot but it's still 7 times slower.

\>`cvt2.cmd` {magick Q8 +dither]
04:40 PM
04:47 PM

**Comment by fmw42** [comment]
ImageMagick and GraphicsMagick are two different tools. GraphicsMagick is smaller and less flexible, so it can run fast. ImageMagick is much more powerful and full-featured and designed for quality. So it is not surprising that it is slower. 

Have timed using magick (IM 7 equivalent of convert) to do your processing. Mogrify is an older IM 6 tool and is being called by IM 7. That could potentially slow it down also. Perhaps writing a script using magick to process your images, might be faster.

**Comment by urban-warrior** [comment]
Thanks for the problem report.  We can reproduce it and will have a patch to fix it in the GIT master branch @ https://github.com/ImageMagick/ImageMagick later today.  The patch will be available in the beta releases of ImageMagick @ https://www.imagemagick.org/download/beta/ by sometime tomorrow.

**Comment by shirha** [comment]
Is there a Windows binary of the above mentioned commit available? I tried ImageMagick-7.0.10-16-Q8-x64-dll.exe but it's not in that one. I was using 7.0.10-14-Q8-x64-dll.
